### DISCUSSION TITLE
Specs (tests) not listed, are there limits over name of spec markdown files?

### DISCUSSION THREAD

**Discussion by rfennell** [testspace-com/support#22] [Q&A]
I have a query over adding TestSpace to a new repo, the tests are now shown

## Setup
I have tried to setup Testspaces on a new repo i.e. one that only contained a `readme.md` and `license`.

I added the `/.testspace.yml` containing 

```
manual:
```
In the `/specs` folder, I added  the most basic spec I could in the file `test a.md`

```
---
testspace:
---

# Q1

## Q1a
```

## What I see
There are no tests show in the TestSpaces UI

![image](https://user-images.githubusercontent.com/324718/131135879-575c5250-f037-4c9a-ad01-528cbc0a333b.png)

But the file is in the repo

![image](https://user-images.githubusercontent.com/324718/131136071-bf976d8f-2cf8-42f5-8c09-d1cf0592cf44.png)

The strange this is that, if I add a new file `/specs/ABC.md` and copy in the content from `/specs/test a.md', the copy shows up in the UI. 

## Questions
This has left me confused, so a couple of questions

- Are the limitation over the names of the spec files?
- Is there a way to see a log that shows the spec file discovery process (as to why a file is not found)?

I am sure it is something I have done wrong, over filename or markdown format

**Comment by munderseth** [comment]
Hi @rfennell I see the specs. Are still not seeing them?

![image](https://user-images.githubusercontent.com/1229367/131139520-83a224cb-1ecb-4562-8bbd-ffb60e80238a.png)

**Reply by rfennell** [reply]
Strangely when I copy files around/rename the files they appear. Seemed to be an issue for just the first specs added. 

When I went through your sample getting started, forked, repo it all worked fine the first time.

It is not an issue, as I am working now. I just asked the question as it made getting started in the empty repo confusing.

**Reply by munderseth** [reply]
Yeah, something seems a little weird. Currently, I can not see your specs on the branch. I forked your repo to look into. 

If you are going to write more specs, use variables, etc., I would recommend using Jekyll for local development and preview. You can get more info here: https://help.testspace.com/manual/implementation-spec#spec-preview.

**Reply by rfennell** [reply]
Thanks for pointer, it is looking a really interesting tool

**Reply by rfennell** [reply]
So you recommend using Jekyll as a preview to make sure the markdown is OK? I had just been using the preview in VSCode.

**Reply by munderseth** [reply]
If you start leveraging [variables](https://help.testspace.com/manual/implementation-spec#variables) and [includes /w parameters](https://help.testspace.com/manual/implementation-spec#includes) the Spec Preview becomes very helpful when designing tests.  Also when writing lots of tests, it can be helpful reviewing them dynamically using a second tab.

**Reply by munderseth** [reply]
Can you try the following (if you have time 😄 ):

- Go to https://github.com/testspace-com/hello.manual and select `Use this template` to create a repo
- Then under your new repo - **Code** - select `Codespaces` (a very nice new feature by GitHub)
- The following the readme instructions - https://github.com/testspace-com/hello.manual#spec-preview

**Reply by rfennell** [reply]
That worked perfectly to allow previewing.

And yes, `Codespaces` are great, perfect for this type of thing. I have been using them all day for my explorations
